test/autoenum-cpp.yaml

```diff
@@ -0,0 +1,10 @@
+domain: cpp
+directive: autoenum
+directive-arguments:
+  - foo
+directive-options:
+  file: enum.c
+  members:
+    - bar
+  clang:
+    - -xc++
```

Comment:
Divided between having `-xc++` and adding similar heuristics as you added in the extension to pass that automatically for cpp domain.

Reply:
Hmm, so default to `-xc++` in `cpp` domain tests and override it with `-xc` for the wrong syntax test? Might be a good idea. Let me think about it, but I think there's no down side.